### PR TITLE
🐛(frontend) manage certificate linked to enrollment

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

Some certificate can be linked to an enrollment. Currently we do not manage
those cases so an error is displayed when we meet this use case.

## Proposal

- [x] Manage certificate linked to enrollment

### CODE REVIEW

File: src/frontend/js/types/Joanie.ts [hunk 1]
```diff
@@ -102,7 +102,8 @@ export interface Certificate {
   id: string;
   issued_on: string;
   certificate_definition: CertificateDefinition;
-  order: NestedCertificateOrder | NestedCredentialOrder;
+  order: Nullable<NestedCertificateOrder | NestedCredentialOrder>;
```
Comment (NathanVss) [inline src/frontend/js/types/Joanie.ts:105]: WDYT about trying to make an exclusion type like "if `order` is defined then `enrollment` is not" and vice-versa? I'm not saying that's easy though 😅

File: src/frontend/js/widgets/Dashboard/components/DashboardItem/Certificate/index.tsx [hunk 2]
```diff
@@ -20,40 +21,43 @@ export const DashboardItemCertificate = ({
   productType,
   mode,
 }: DashboardItemCertificateProps) => {
-  if (certificate) {
-    if (certificateDefinition) {
-      throw new Error('certificate and certificateDefinition are mutually exclusive');
+  const getCourseFromCertificate = () => {
```
Comment (NathanVss) [inline src/frontend/js/widgets/Dashboard/components/DashboardItem/Certificate/index.tsx:24]: In the future we will maybe have to make this function available to other components, WDYT about make a standalone function? And then writing a unit test forit?

**Review by NathanVss — APPROVED** [review]
📜

File: src/frontend/js/widgets/Dashboard/components/DashboardItem/Certificate/index.tsx (hunk 2 above)
Comment (jbpenrath) [reply src/frontend/js/widgets/Dashboard/components/DashboardItem/Certificate/index.tsx]: Ok, I created a `CertificateHelper` implementing a `getCourse` method. Let's take a look to the dedicated commit.

File: src/frontend/js/types/Joanie.ts (hunk 1 above)
Comment (jbpenrath) [reply src/frontend/js/types/Joanie.ts]: Please review the fixup commit to check if the proposed solution is fine for you.

**Pull request merged**